.travis.yml

```diff
@@ -108,13 +108,10 @@ after_script: skip
     #- find ./rpm -iname '*.rpm' -o -iname '*.tar.gz' -o -iname '*.tbz2'
   after_success:
     - echo "Testing the installation of the package"
-    - virtualenv building -p python --system-site-packages
-    - . building/bin/activate
-    - pip install python/reg_interface/rpm/*.tar.gz
-    - echo "Generate documentation"
-    - export REG_XML_PATH=$PWD
-    - export XHAL_ROOT=$PWD
-    - make doc
+    - echo "First install rwreg"
+    - docker exec -ti $DOCKER_CONTAINER_ID /bin/bash -ec "cd /home/daqbuild/${REPO_NAME}; sudo rpm -ivh rwreg/x86/rpm/*.rpm"
+    - echo "Now install reg_interface and generate documentation"
+    - docker exec -ti $DOCKER_CONTAINER_ID /bin/bash -ec "cd /home/daqbuild; virtualenv building -p python --no-site-packages; . building/bin/activate; cd ${REPO_NAME}; pip install pip==9.0.3; pip install --upgrade setuptools; pip install sphinx; pip install python/reg_interface/rpm/*.tar.gz; export REG_XML_PATH=$PWD; export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/opt/rwreg/lib:/opt/wiscrpcsvc/lib; make doc"
```

Comment:
at some future point i'd probably put this stuff into a test/docbuild script, as it would be directly migrateable to a `gitlab-ci` workflow

Reply:
> Can you check the filesystem structure of the built RPM from this env?
> (rpm -q --qf "[%-15{=NAME} %-36{FILENAMES} %{FILEMODES:perms}\n]" -p <rpm file>)
> My suspicion/fear is that the built RPM will explicitly require rh/python27 to be enabled, and > will install into that environment
> Also, this shouldn't really be necessary except in the event that the system python is not python2.7

I'm actually not sure whether that sourcing is required. I commented it out as the sourcing is only required to build the docs. Everything else should build just fine with system python. 

> at some future point i'd probably put this stuff into a test/docbuild script, as it would be directly migrateable to a gitlab-ci workflow

I totally agree. However right now this is the only way to save some time during the build and do not pull everything for the new docker image from scratch. 

Actually the biggest point is: letting some small details aside, a set of rpms is produced and when installed under correct python environment, everything is imported as expected. So I think this PR can be merged and we can start manual testing to see whether everything will work properly when we deploy it to the real PC. The next steps should be:
* add packaging to xhal - should be reasonably easy now, as we have all the ingredients
* modify imports in the files recenly committed by Brian
* test everything on the host
* prepare rpms for ARM

Reply:
Actually I should say there's still something fishy with the doc build under slc6 and it still uses `python 2.6.6` despite command to build the venv with `rh` python... I believe this can be sorted later (eventually when we migrate to gitlab) and in principle it is rather a problem of PC setup.